test/jit/test_union.py

```diff
@@ -334,12 +378,32 @@ def forward(self, z: str):
 
         self.checkModule(M(1), ("foo",))
 
+    def test_union_subtractive_refinement(self):
+        def fn(x: Union[List[int], int]):
```

Comment:
This test case doesn't really test that `x` in else branch is an int because `fn` doesn't have a clear return type declared.
```suggestion
        def fn(x: Union[List[int], int]) -> int:
```